pkg/client/events_tracking_test.go

```diff
@@ -0,0 +1,361 @@
+package client
+
+import (
+	"encoding/hex"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func TestRequestedSignersTrackAdd(t *testing.T) {
+	keepAddress1 := common.BytesToAddress([]byte{1})
+	keepAddress2 := common.BytesToAddress([]byte{2})
+
+	rs := &requestedSignersTrack{
+		data:  make(map[string]bool),
+		mutex: &sync.Mutex{},
+	}
+
+	if !rs.add(keepAddress1) {
+		t.Error("unexpected failure")
+	}
+
+	if ok := rs.data[keepAddress1.String()]; !ok {
+		t.Errorf(
+			"unexpected value for keep [%s]\nexpected: [%v]\nactual:   [%v]",
+			keepAddress1.String(),
+			true,
+			rs.data[keepAddress1.String()],
+		)
+	}
+
+	if ok := rs.add(keepAddress2); !ok {
+		t.Error("unexpected failure")
+	}
+
+	if !rs.data[keepAddress2.String()] {
+		t.Errorf(
+			"unexpected value for keep [%s]\nexpected: [%v]\nactual:   [%v]",
+			keepAddress2.String(),
+			true,
+			rs.data[keepAddress1.String()],
+		)
+	}
+}
+
+func TestRequestedSignersTrackAdd_Duplicate(t *testing.T) {
+	keepAddress1 := common.BytesToAddress([]byte{1})
+
+	rs := &requestedSignersTrack{
+		data:  make(map[string]bool),
+		mutex: &sync.Mutex{},
+	}
+
+	if ok := rs.add(keepAddress1); !ok {
+		t.Error("unexpected failure")
+	}
+
+	if ok := rs.add(keepAddress1); ok {
+		t.Errorf(
+			"unexpected result\nexpected: [%v]\nactual:   [%v]",
+			false,
+			ok,
+		)
```

Comment:
In the previous assertion we don't print expected vs actual, I think we should use the same pattern here and say e.g. `signature already requested` or something similar.

Reply:
For consistency compare whole data map? (https://github.com/keep-network/keep-ecdsa/pull/372#discussion_r405451898)